mlir-sycl/lib/Dialect/SYCL/Transforms/ConstantPropagation.cpp

```diff
@@ -1149,6 +1146,7 @@ void ConstantPropagationPass::propagateImplicitConstantArgs(
 }
 
 void ConstantPropagationPass::runOnOperation() {
+  SymbolTableCollection symbolTable;
```

Comment:
We can use this no problem as no relevant symbol change will be made between kernels